clipper_admin/clipper_admin/clipper_admin.py

```diff
@@ -1187,3 +1190,68 @@ def stop_all(self):
         """
         self.cm.stop_all()
         logger.info("Stopped all Clipper cluster and all model containers")
+
+    def test_predict_function(self, query, func, input_type):
+        """Tests that the user's function has the correct signature and can be properly saved and loaded.
+
+        The function should take a dict request object like the query frontend expects JSON, 
+        the predict function, and the input type for the model.
+
+        Parameters
+        ----------
+        query: JSON or list of dicts
+            Inputs to test the prediction function on.
+        func: function
+            Predict function to test.
+        input_type: str
+            The input_type to be associated with the registered app and deployed model.
+            One of "integers", "floats", "doubles", "bytes", or "strings".
+        """
+        query_data = list(x for x in list(query.values()))
+        
+        if type(query_data[0][0]) == list:
+            query_data = query_data[0]
```

Comment:
What are you checking for here?

Reply:
I'm checking the nesting of the query data, whether it is a list or a list of lists.